### DISCUSSION TITLE
mbed-tools and custom target.

### DISCUSSION THREAD

**Discussion by indyIOT** [ARMmbed/mbed-tools#276] [Q&A]
I have been watching, the repo to see when custom targets would be available. They seem to be now, but when I use a custom target the configure generates proper output. But when I use the cmake command I get a no cmakelists file. Do I need to generate one? Or is the tool supposed to? If the tool does where does it put it. 

The source directory "/root/devops_firmware/Project-Root" does not appear to contain CMakeLists.txt.

mbed_config.cmake has been generated and written to '/root/devops_firmware/Project-Name/cmake_build/PROJECT-NAME_2M/develop/GCC_ARM/mbed_config.cmake'

**Comment by Patater** [comment]
Hi @tbandtg 

Yes, adding new targets requires writing a CMakeLists.txt for you. The documentation for how to make the CMakeLists.txt file (and others needed for custom targets) are at https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/porting/custom-target-porting.md
